container/container.go

```diff
@@ -667,15 +667,30 @@ func (container *Container) BuildCreateEndpointOptions(n libnetwork.Network, epC
 
 	if epConfig != nil {
 		ipam := epConfig.IPAMConfig
-		if ipam != nil && (ipam.IPv4Address != "" || ipam.IPv6Address != "" || len(ipam.LinkLocalIPs) > 0) {
+
+		if ipam != nil {
 			var ipList []net.IP
+			var ip, ip6, linkip net.IP
+
 			for _, ips := range ipam.LinkLocalIPs {
-				if ip := net.ParseIP(ips); ip != nil {
-					ipList = append(ipList, ip)
+				if linkip = net.ParseIP(ips); linkip == nil && ips != "" {
+					return nil, fmt.Errorf("Error parsing parameter link-local-ip value(%s):%s", ipam.LinkLocalIPs, ips)
 				}
+				ipList = append(ipList, linkip)
+
+			}
+
+			if ip = net.ParseIP(ipam.IPv4Address); ip == nil && ipam.IPv4Address != "" {
+				return nil, fmt.Errorf("Error parsing parameter ip value(%s)", ipam.IPv4Address)
```

Comment:
It would be nice to conform these errors to the errors docker already returns for invalid IP on `docker run`
As an example, I see
`invalid IPv4 address: 3.3.3.333`
`invalid IPv6 address: 2001:db8:12345678::1`